addOns/reports/src/main/java/org/zaproxy/addon/reports/sarif/SarifBinaryContentDetector.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.reports.sarif;
+
+import static java.util.Objects.requireNonNull;
+
+import org.parosproxy.paros.network.HttpHeader;
+
+public class SarifBinaryContentDetector {
+
+    public static final SarifBinaryContentDetector DEFAULT = new SarifBinaryContentDetector();
+
+    /**
+     * Detect binary content by inspecting the given HTTP header and the returned normalized
+     * (lowercased) content type. see <a
+     * href="https://www.iana.org/assignments/media-types/media-types.xhtml">IANA Mime type list</a>
+     * for a list of mime types.
+     *
+     * @param header the HTTP header to inspect
+     * @return <code>true</code> when binary content otherwise <code>false</code>
+     */
+    public boolean isBinaryContent(HttpHeader header) {
```

Comment:
Note: commonlib does have a method for checking whether or not responses contain control chars, which may be of benefit here? (If the actual content, not just the header, is available.)

https://github.com/zaproxy/zap-extensions/blob/d7b0b875a37f40cbf3d6c95d9054e5e232128614/addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ResourceIdentificationUtils.java#L112-L117

Reply:
Hm.. interesting, but I think -  at least as a first version of the SARIF report handling - we should rely on the content type returned by the server. I am not sure if the existence of an ISO control character is really the 100% way to accept the content as a binary or not. To avoid edge cases, I would keep it as is.

Reply:
That’s fine